### PR TITLE
feat: emotes catalog

### PR DESCRIPTION
**Pull request by AjimenezDCL**
This PR prepares everything for NFT Emotes #1833.
It lays down the foundation to work with a catalog of animations without replacing the current embedded emotes. Bear in mind that some of the code to handle the embedded emotes is temporary (although it shouldnt be nothing too egregious).

## Context
`EmoteAnimationsPlugin`: Handles the retrieval of the _AnimationClips_, which by now is just adding the embedded ones into the DataStore (notice how the methods to actually load remote Emotes are not impemented yet).

`EmoteAnimationEquipper`: Equips the emotes animations when they are ready.

The flow goes as follow:
1) User profile received containing emotes (included in the _avatar.wearables_ list)
2) `AvatarSystem` and specifically the `AvatarCurator` resolves ids to `WearableItem` and split wearables from emotes.
3) `EmoteAnimationEquipper` receives the list of emotes and equip the ones which _AnimationClip_ is ready.
4) `EmoteAnimationsPlugin` detects the usage of the equipped emotes and downloads them, adding the _AnimationClip_ to the DataStore.
5) `EmoteAnimationEquipper` detects the new animations ready and equips them in the `Animator`.

Currently Step 4 and 5 are not taking place since we only work with embedded emotes by now.

Since the user profile doesnt carry emotes yet, all the embedded ones are injected before loading the avatar. This is, of course, temporary.

## How to Test
There's no new features. Take a look at avatars and specifically to emotes. They should behave as in production.

### CODE REVIEW

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Components/Avatar/AvatarAnimatorLegacy.cs [hunk 1]
```diff
@@ -302,12 +290,28 @@ public bool Prepare(string bodyshapeId, GameObject container)
...
-        BindBodyShape(animation, bodyshapeId, target);
+        PrepareLocomotionAnims(bodyshapeId);
+        SetIdleFrame();
+        animation.Sample();
         InitializeAvatarAudioAndParticleHandlers(animation);
         return true;
     }
 
-    public void PlayExpression(string expressionId, long timestamps) { SetExpressionValues(expressionId, timestamps); }
+    public void PlayEmote(string emoteId, long timestamps) { SetExpressionValues(emoteId, timestamps); }
```
Comment (BrianAmadori) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Components/Avatar/AvatarAnimatorLegacy.cs:300]: There's an ongoing confusion organization-wide in which we are calling all the animations as emotes. The emotes concept comes with the burden of being a fire and forget animation that is tied with the emotes UI, marketplace, etc. However, here we are indirectly calling "emotes" to the locomotion animations that do not have this burden and are not tied to any UI or marketplace.

Maybe we can call those methods `EquipAnimation` or `BindAnimation` to appease this concern?

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/CatalogController/CatalogController.cs [hunk 2]
```diff
@@ -360,6 +375,7 @@ private void CheckForUnusedWearables()
 
             foreach (var wearableToDestroy in wearablesToDestroy)
             {
+                Debug.Log($"Removing: {wearableToDestroy}");
```
Comment (BrianAmadori) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/CatalogController/CatalogController.cs:378]:
```suggestion
```

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/PlayerAvatarController/PlayerAvatarController.cs [hunk 3]
```diff
@@ -120,7 +123,7 @@ private void OnEnable()
 
     private void OnAvatarExpression(string id, long timestamp)
```
Comment (BrianAmadori) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/PlayerAvatarController/PlayerAvatarController.cs:124]:
```suggestion
    private void OnAvatarEmote(string id, long timestamp)
```

File: unity-renderer/Assets/Scripts/MainScripts/DCL/AvatarSystem/Avatar.cs [hunk 4]
```diff
@@ -87,7 +95,10 @@ public async UniTask Load(List<string> wearablesIds, AvatarSettings settings, Ca
             {
                 Dispose();
                 Debug.Log($"Avatar.Load failed with wearables:[{string.Join(",", wearablesIds)}] for bodyshape:{settings.bodyshapeId} and player {settings.playerName}");
-                throw;
+                if (e.InnerException != null)
+                    ExceptionDispatchInfo.Capture(e.InnerException).Throw();
```
Comment (BrianAmadori) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/AvatarSystem/Avatar.cs:99]: What's this for?

**Review by sandrade-dcl — APPROVED** [review]

**Pull request merged**